pkgs/development/python-modules/furl/default.nix

```
@@ -36,10 +36,30 @@ buildPythonPackage rec {
     pytestCheckHook
   ];
 
-  disabledTests = lib.optionals (pythonAtLeast "3.12") [
-    # AssertionError: assert '//////path' == '////path'
-    "test_odd_urls"
-  ];
+  disabledTests =
+    # test failure "test_odd_urls"
+    # https://github.com/gruns/furl/issues/176
+    let
+      pythonVersionFull = with python.sourceVersion; "${major}.${minor}.${patch}";
+      pythonAtLeast' =
+        v: (python.pythonVersion == lib.versions.majorMinor v) && (lib.versionAtLeast pythonVersionFull v);
+      cond = (
+        lib.lists.any lib.trivial.id (
+          lib.map pythonAtLeast' [
+            "3.13.0"
+            "3.12.4"
+            "3.11.10"
+            "3.10.15"
+            "3.9.20"
+            "3.8.20"
+          ]
+        )
+      );
+    in
+    lib.optionals cond [
```

Let's make this unconditional.
```suggestion
    [
```

---

make that test to be disabled always?

---

Yes

---

Since we know for sure these are the versions where it was introduced, isn't this version checking logic good enough?

once furl gets a new version then the it can be removed.

---

But we have only those versions of Python.

---

In master I remember atleast one of the pythons versions being lower, I guess in staging, they're all there now.
Ok then, I'll remove it.

Another test with websockets package is failing in staging, I'll ping for review again.